src/wallet/wallet.cpp

```diff
@@ -784,6 +784,27 @@ bool CWallet::IsSpentKey(const uint256& hash, unsigned int n) const
     return false;
 }
 
+void CWallet::SetUsedAddressState(WalletBatch& batch, const CTxOut& cout, bool used)
+{
+    AssertLockHeld(cs_wallet);
+    CTxDestination dst;
+    if (ExtractDestination(cout.scriptPubKey, dst)) {
+        if (IsMine(dst)) {
+            if (used && !GetDestData(dst, "address_used", nullptr)) {
+                AddDestData(batch, dst, "address_used", "p"); // p for "present", opposite of absent (null)
+            } else if (!used && GetDestData(dst, "address_used", nullptr)) {
+                EraseDestData(batch, dst, "address_used");
+            }
+        }
+    }
+}
```

Comment:
perhaps simplify
```diff
 {
     AssertLockHeld(cs_wallet);
     CTxDestination dst;
-    if (ExtractDestination(cout.scriptPubKey, dst)) {
-        if (IsMine(dst)) {
-            if (used && !GetDestData(dst, "address_used", nullptr)) {
-                AddDestData(batch, dst, "address_used", "p"); // p for "present", opposite of absent (null)
-            } else if (!used && GetDestData(dst, "address_used", nullptr)) {
-                EraseDestData(batch, dst, "address_used");
-            }
+    if (ExtractDestination(cout.scriptPubKey, dst) && IsMine(dst)) {
+        if (used && !GetDestData(dst, "address_used", nullptr)) {
+            AddDestData(batch, dst, "address_used", "p"); // p for "present", opposite of absent (null)
+        } else if (!used && GetDestData(dst, "address_used", nullptr)) {
+            EraseDestData(batch, dst, "address_used");
         }
     }
 }
```

Reply:
Yes makes sense, this has now been simplified as suggested